tests/apis/parameters.spec.ts

```diff
@@ -0,0 +1,94 @@
+import * as vscode from "vscode";
+import { expect } from "chai";
+import { getParameter } from '../../src/apis/parameters';
+
+describe("getParameter API", () => {
+    const parameterName = "param1";
+
+    context("no @sap/plugin loaded", () => {
+        it("should return undefined", async () => {
+            const parameterValue = await getParameter(parameterName);
+            expect(parameterValue).to.be.undefined;
+        });
+    });
+
+    context("@sap/plugin loaded", () => {
+        // no test for configuration is undefined, because . behaves the same on null and undefined
+        describe("when configuration is null", () => { 
+            let requireMock;
+            
+            before(() => {
+                requireMock = require('mock-require');
+                const sapPlugin = {
+                    window: {
+                        configuration: () => null
+                    }
+                };
+                requireMock('@sap/plugin', sapPlugin);
+            })
+
+            it("should return undefined", async () => {
+                const parameterValue = await getParameter(parameterName);
+                expect(parameterValue).to.be.undefined;
+            });
+            
+            after(() => {
+                requireMock.stop('@sap/plugin');
+            });
+        });
+    });
+      
+    context("@sap/plugin loaded", () => {
+        // no test for configuration containing other parameters, because [] behaves the same

```

Comment:
[] --> to 'member access'

because `[]` can also reference an array or destructuring in JavaScript syntax.